io/zenoh-transport/src/common/pipeline.rs

```diff
@@ -140,19 +148,24 @@ impl StageIn {
                             }
                             None => {
                                 drop(c_guard);
-                                if !$fragment && is_droppable {
-                                    // Restore the sequence number
-                                    $restore_sn;
-                                    // We are in the congestion scenario
-                                    // The yield is to avoid the writing task to spin
-                                    // indefinitely and monopolize the CPU usage.
-                                    thread::yield_now();
-                                    return false;
-                                } else {
-                                    if !self.s_ref.wait() {
+                                match deadline_before_drop {
+                                    Some(deadline) if !$fragment => {
                                         // Restore the sequence number
                                         $restore_sn;
```

Comment:
`$restore_sn` should be called only if we are dropping the message, i.e. inside the 

```rust
if !self.s_ref.wait_deadline(deadline) {
    $restore_sn
    return false;
}
```

Reply:
Fixed